### PR TITLE
Allow alternative names for overridable PSA headers

### LINKED ISSUE #7145: Allow alternative names for crypto_struct.h and crypto_platform.h
The headers `crypto_platform.h` and `crypto_struct.h` are carved out so that integrators can override our version with their own. For `crypto_struct.h`, this is intended for the client side in a client-server integration. Overriding `crypto_platform.h` is useful in several cases including client-server integrations.

Currently, you can override those headers either by overwriting the files in the source tree, or by shadowing them through the include directory search path. Sometimes this is inconvenient. In particular, this is inconvenient for TF-M.

The goal of this issue is to allow overriding `psa/crypto_platform.h` and `psa/crypto_struct.h` through preprocessor symbols, the same way you can override `mbedtls/mbedtls_config.h` and `psa/crypto_config.h`.

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Allow alternative paths for the PSA headers that integrators may override (`psa/crypto_platform.h` and `psa/crypto_struct.h`). Fixes #7145.

Also move the implication `MBEDTLS_PSA_CRYPTO_C` ⇒ `MBEDTLS_PSA_CRYPTO_CLIENT` to `build_info.h` where it belongs, in partial fulfilment of https://github.com/Mbed-TLS/mbedtls/issues/7144.

## Gatekeeper checklist

- [x] **changelog** provided
- [x] **backport** no (new feature)
- [x] **tests** provided

### CODE REVIEW

**Comment by yuhaoth** [comment]
I just take a quick look on this one. and it looks have some next PRs.  Maybe I am wrong.

I'd like enhance `query_config` or rename it to `query_psa_config` .  PSA is more like a crypto lib  depended by MBedtls. To prepare that, we should use a name that can show the ideas.

Beside that, I prefer merge it into `query_config` , make it platform independent and become part of library release.

File: include/mbedtls/mbedtls_config.h [hunk 1]
```diff
@@ -3537,6 +3537,53 @@
  */
 //#define MBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE "/dev/null"
 
+/**
```
Comment (yuhaoth) [inline include/mbedtls/mbedtls_config.h:3540]: I do not think it should be added in `mbedtls_config.h`. 

I think it should be move to `psa_config.h` ( I am not sure the real name at this moment) or something like this.

**Review by AndrzejKurek — APPROVED** [review]
Looks good to me.

**Review by yuhaoth — APPROVED** [review]
LGTM

**Comment by daverodgman** [comment]
CI issues are unrelated

**Pull request merged**